test/over_react/component_declaration/builder_integration_tests/component2/constant_required_accessor_integration_test.dart

```diff
@@ -23,137 +22,114 @@ part 'constant_required_accessor_integration_test.over_react.g.dart';
 
 void main() {
   group('(Component2) propTypes properly identifies required props by', () {
-    List<String> consoleErrors;
-    JsFunction originalConsoleError;
-
-    setUp(() {
-      // PropTypes by default will only throw a specific error one time per Component Class.
-      // This resets the cache after each test so it throws again.
-      // See: https://www.npmjs.com/package/prop-types#proptypesresetwarningcache
-      PropTypes.resetWarningCache();
-
-      originalConsoleError = context['console']['error'];
-      consoleErrors = [];
-      context['console']['error'] = JsFunction.withThis((self, message) {
-        consoleErrors.add(message);
-        originalConsoleError.apply([message], thisArg: self);
-      });
-    });
-
-    tearDown(() {
-      context['console']['error'] = originalConsoleError;
-    });
     group('throwing when a prop is required and not set', () {
-
       test('on mount', () {
-        mount(
-          (ComponentTest()
-            ..nullable = true
-            ..requiredAndLengthLimited = [1,2]
-          )(),
-          attachedToDocument: true,
-        );
-
-        expect(consoleErrors, isNotEmpty, reason: 'should have outputted a warning');
-        expect(consoleErrors, [contains('ComponentTestProps.required')]);
+        expect(() {
+          mount(
+            (ComponentTest()
+              ..nullable = true
+              ..requiredAndLengthLimited = [1,2]
+            )(),
+            attachedToDocument: true,
+          );
+        }, logsPropRequiredError('ComponentTestProps.required'));
       });
 
       test('on re-render', () {
-        var jacket = mount((ComponentTest()
-            ..required = true
-            ..nullable = true
-            ..requiredAndLengthLimited = [1,2]
-          )(),
-          attachedToDocument: true,
-        );
-
-        expect(consoleErrors, isEmpty, reason: 'should not have outputted a warning but found: $consoleErrors');
-
-        jacket.rerender((ComponentTest()
+        TestJacket jacket;
+
+        expect(() {
+          jacket = mount(
+              (ComponentTest()
+                ..required = true
+                ..nullable = true
+                ..requiredAndLengthLimited = [1,2])(),
```

Comment:
#nit component formatting
```suggestion
                ..requiredAndLengthLimited = [1,2]
              )(),
```